core/slice.go

```diff
@@ -1005,15 +1034,78 @@ func (sl *Slice) pcrc(batch ethdb.Batch, header *types.WorkObject, domTerminus c
 }
 
 // GetPendingHeader is used by the miner to request the current pending header
-func (sl *Slice) GetPendingHeader() (*types.WorkObject, error) {
+func (sl *Slice) GetPendingHeader(powId types.PowID, coinbase common.Address) (*types.WorkObject, error) {
```

Comment:
Should this be in slice or headerchain?

Reply:
slice because headerchain doesnt have access to the worker